```
Error running mint automation
+ mkdir -p 8307-bb0523d/gopath/src/github.com/minio
+ git -C 8307-bb0523d/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 8307-bb0523d/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 8307-bb0523d/gopath/src/github.com/minio/minio fetch --quiet minio pull/8307/head:pr8307
+ git -C 8307-bb0523d/gopath/src/github.com/minio/minio checkout --quiet pr8307
+ GO111MODULE=on
+ GOPATH=/mnt/nvme/mint-auto/8307-bb0523d/gopath
+ make -C 8307-bb0523d/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/minio/cli v1.21.0
go: downloading github.com/Azure/go-autorest v11.7.0+incompatible
go: downloading github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: downloading github.com/minio/hdfs/v3 v3.0.1
go: downloading github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: downloading github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: downloading github.com/json-iterator/go v1.1.7
go: downloading cloud.google.com/go v0.37.2
go: extracting github.com/json-iterator/go v1.1.7
go: downloading google.golang.org/api v0.4.0
go: extracting github.com/minio/cli v1.21.0
go: extracting github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: downloading github.com/cheggaaa/pb v1.0.28
go: extracting github.com/minio/hdfs/v3 v3.0.1
go: downloading github.com/eclipse/paho.mqtt.golang v1.1.2-0.20190322152051-20337d8c3947
go: downloading github.com/coreos/etcd v3.3.12+incompatible
go: extracting github.com/Azure/go-autorest v11.7.0+incompatible
go: downloading go.uber.org/atomic v1.3.2
go: extracting github.com/cheggaaa/pb v1.0.28
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: extracting github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: downloading github.com/minio/gokrb5/v7 v7.2.5
go: extracting github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: extracting github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: extracting go.uber.org/atomic v1.3.2
go: downloading gopkg.in/Shopify/sarama.v1 v1.20.0
go: downloading github.com/minio/minio-go/v6 v6.0.37
go: extracting github.com/eclipse/paho.mqtt.golang v1.1.2-0.20190322152051-20337d8c3947
go: downloading github.com/gorilla/rpc v1.2.0
go: extracting cloud.google.com/go v0.37.2
go: downloading github.com/minio/dsync v1.0.0
go: extracting github.com/coreos/etcd v3.3.12+incompatible
go: extracting github.com/minio/minio-go/v6 v6.0.37
go: extracting github.com/gorilla/rpc v1.2.0
go: downloading golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: downloading github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: downloading github.com/minio/mc v0.0.0-20190924013003-643835013047
go: downloading github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: downloading github.com/minio/lsync v1.0.1
go: extracting gopkg.in/Shopify/sarama.v1 v1.20.0
go: extracting github.com/minio/dsync v1.0.0
go: downloading github.com/minio/dsync/v2 v2.0.0
go: extracting golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: extracting github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: extracting google.golang.org/api v0.4.0
go: downloading golang.org/x/crypto v0.0.0-20190923035154-9ee001bba392
go: extracting github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: downloading github.com/djherbis/atime v1.0.0
go: extracting github.com/minio/lsync v1.0.1
go: downloading google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: extracting github.com/minio/mc v0.0.0-20190924013003-643835013047
go: extracting github.com/minio/dsync/v2 v2.0.0
go: downloading github.com/klauspost/pgzip v1.2.1
go: downloading github.com/nsqio/go-nsq v1.0.7
go: downloading github.com/mattn/go-colorable v0.1.1
go: downloading github.com/dustin/go-humanize v1.0.0
go: extracting github.com/djherbis/atime v1.0.0
go: downloading github.com/pkg/profile v1.3.0
go: extracting golang.org/x/crypto v0.0.0-20190923035154-9ee001bba392
go: downloading github.com/miekg/dns v1.1.8
go: extracting github.com/dustin/go-humanize v1.0.0
go: downloading github.com/go-sql-driver/mysql v1.4.1
go: extracting github.com/mattn/go-colorable v0.1.1
go: downloading github.com/nats-io/nats.go v1.8.0
go: extracting github.com/nsqio/go-nsq v1.0.7
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: extracting github.com/klauspost/pgzip v1.2.1
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: extracting github.com/go-sql-driver/mysql v1.4.1
go: extracting github.com/pkg/profile v1.3.0
go: downloading golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: extracting github.com/miekg/dns v1.1.8
go: downloading google.golang.org/grpc v1.20.1
go: extracting github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/klauspost/readahead v1.3.0
go: extracting github.com/nats-io/nats.go v1.8.0
go: downloading github.com/nats-io/nkeys v0.0.2
go: extracting github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading github.com/prometheus/common v0.2.0
go: extracting golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: downloading github.com/minio/highwayhash v1.0.0
go: extracting github.com/nats-io/nkeys v0.0.2
go: extracting github.com/klauspost/readahead v1.3.0
go: downloading gopkg.in/ini.v1 v1.48.0
go: extracting github.com/minio/highwayhash v1.0.0
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: extracting github.com/prometheus/common v0.2.0
go: extracting golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: downloading github.com/golang/snappy v0.0.1
go: extracting github.com/golang/snappy v0.0.1
go: downloading github.com/golang/protobuf v1.3.1
go: extracting google.golang.org/grpc v1.20.1
go: extracting github.com/golang/protobuf v1.3.1
go: downloading github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: extracting gopkg.in/ini.v1 v1.48.0
go: downloading github.com/klauspost/compress v1.5.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: extracting github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/hashicorp/vault v1.1.0
go: extracting github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: extracting github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/minio/sio v0.2.0
go: extracting github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: extracting github.com/minio/gokrb5/v7 v7.2.5
go: downloading github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: extracting github.com/minio/sio v0.2.0
go: downloading go.opencensus.io v0.21.0
go: extracting github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: downloading github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: extracting google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: extracting github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading github.com/gorilla/mux v1.7.0
go: extracting go.opencensus.io v0.21.0
go: extracting github.com/gorilla/mux v1.7.0
go: downloading github.com/gogo/protobuf v1.2.1
go: downloading github.com/googleapis/gax-go v2.0.0+incompatible
go: extracting github.com/googleapis/gax-go v2.0.0+incompatible
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: extracting github.com/googleapis/gax-go/v2 v2.0.4
go: downloading github.com/minio/sha256-simd v0.1.1
go: extracting github.com/minio/sha256-simd v0.1.1
go: downloading github.com/pkg/errors v0.8.1
go: extracting github.com/pkg/errors v0.8.1
go: downloading github.com/Azure/azure-sdk-for-go v27.0.0+incompatible
go: downloading github.com/hashicorp/golang-lru v0.5.1
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/lib/pq v1.0.0
go: downloading gopkg.in/olivere/elastic.v5 v5.0.80
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: extracting github.com/hashicorp/golang-lru v0.5.1
go: downloading github.com/fatih/structs v1.1.0
go: extracting github.com/rjeczalik/notify v0.9.2
go: downloading github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: extracting github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/hashicorp/go-uuid v1.0.1
go: extracting github.com/lib/pq v1.0.0
go: extracting github.com/nats-io/nuid v1.0.1
go: downloading github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: downloading github.com/davecgh/go-spew v1.1.1
go: extracting github.com/gogo/protobuf v1.2.1
go: downloading github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: extracting github.com/fatih/structs v1.1.0
go: downloading github.com/mattn/go-runewidth v0.0.4
go: extracting gopkg.in/olivere/elastic.v5 v5.0.80
go: extracting github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: extracting github.com/hashicorp/go-uuid v1.0.1
go: downloading github.com/fatih/color v1.7.0
go: downloading github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: extracting github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: downloading github.com/alecthomas/participle v0.2.1
go: extracting github.com/davecgh/go-spew v1.1.1
go: downloading github.com/tidwall/gjson v1.2.1
go: extracting github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: extracting github.com/matttproud/golang_protobuf_extensions v1.0.1
go: extracting github.com/fatih/color v1.7.0
go: downloading github.com/gorilla/handlers v1.4.0
go: extracting github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: downloading golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: extracting github.com/alecthomas/participle v0.2.1
go: extracting github.com/tidwall/gjson v1.2.1
go: downloading github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: extracting gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading github.com/nats-io/stan.go v0.4.5
go: downloading github.com/ncw/directio v1.0.5
go: extracting github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: downloading gopkg.in/yaml.v2 v2.2.2
go: extracting github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: extracting github.com/gorilla/handlers v1.4.0
go: downloading contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: extracting github.com/ncw/directio v1.0.5
go: extracting github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: downloading github.com/coredns/coredns v1.4.0
go: extracting github.com/nats-io/stan.go v0.4.5
go: downloading github.com/klauspost/reedsolomon v1.9.1
go: extracting gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: extracting gopkg.in/yaml.v2 v2.2.2
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading gopkg.in/ldap.v3 v3.0.3
go: extracting contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: downloading golang.org/x/text v0.3.2
go: extracting golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: extracting github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/census-instrumentation/opencensus-proto v0.2.0
go: downloading github.com/pierrec/lz4 v2.0.5+incompatible
go: downloading github.com/rs/cors v1.6.0
go: extracting github.com/coredns/coredns v1.4.0
go: downloading github.com/mattn/go-isatty v0.0.7
go: extracting gopkg.in/ldap.v3 v3.0.3
go: extracting github.com/mattn/go-isatty v0.0.7
go: downloading github.com/eapache/queue v1.1.0
go: extracting github.com/census-instrumentation/opencensus-proto v0.2.0
go: downloading gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: downloading github.com/tidwall/pretty v1.0.0
go: extracting github.com/rs/cors v1.6.0
go: extracting github.com/klauspost/reedsolomon v1.9.1
go: downloading github.com/eapache/go-resiliency v1.2.0
go: extracting github.com/tidwall/pretty v1.0.0
go: extracting github.com/eapache/queue v1.1.0
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: extracting gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: extracting github.com/eapache/go-resiliency v1.2.0
go: downloading golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: extracting gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading github.com/tidwall/match v1.0.1
go: extracting github.com/pierrec/lz4 v2.0.5+incompatible
go: extracting gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: downloading github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: extracting golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: extracting github.com/tidwall/match v1.0.1
go: downloading github.com/klauspost/cpuid v1.2.1
go: extracting github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: downloading git.apache.org/thrift.git v0.12.0
go: extracting github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: extracting github.com/klauspost/cpuid v1.2.1
go: extracting golang.org/x/text v0.3.2
go: extracting git.apache.org/thrift.git v0.12.0
go: extracting github.com/klauspost/compress v1.5.0
go: extracting github.com/hashicorp/vault v1.1.0
go: extracting github.com/Azure/azure-sdk-for-go v27.0.0+incompatible
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/hashicorp/go-retryablehttp v0.5.3
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: extracting github.com/hashicorp/go-cleanhttp v0.5.1
go: extracting github.com/hashicorp/errwrap v1.0.0
go: extracting github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/ryanuber/go-glob v1.0.0
go: extracting github.com/mitchellh/mapstructure v1.1.2
go: extracting github.com/hashicorp/go-sockaddr v1.0.2
go: extracting github.com/ryanuber/go-glob v1.0.0
go: extracting github.com/hashicorp/go-rootcerts v1.0.1
go: extracting github.com/hashicorp/go-retryablehttp v0.5.3
go: extracting github.com/hashicorp/hcl v1.0.0
go: downloading github.com/satori/go.uuid v1.2.0
go: downloading github.com/marstr/guid v1.1.0
go: extracting github.com/satori/go.uuid v1.2.0
go: extracting github.com/marstr/guid v1.1.0
go: finding github.com/minio/gokrb5/v7 v7.2.5
go: finding github.com/minio/minio-go/v6 v6.0.37
go: finding github.com/dustin/go-humanize v1.0.0
go: finding github.com/minio/cli v1.21.0
go: finding github.com/minio/hdfs/v3 v3.0.1
go: finding github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: finding github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: finding github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: finding github.com/fatih/color v1.7.0
go: finding github.com/Azure/go-autorest v11.7.0+incompatible
go: finding github.com/json-iterator/go v1.1.7
go: finding github.com/minio/sha256-simd v0.1.1
go: finding cloud.google.com/go v0.37.2
go: finding github.com/minio/mc v0.0.0-20190924013003-643835013047
go: finding github.com/mattn/go-isatty v0.0.7
go: finding github.com/mattn/go-colorable v0.1.1
go: finding google.golang.org/api v0.4.0
go: finding github.com/minio/highwayhash v1.0.0
go: finding github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: finding github.com/gorilla/mux v1.7.0
go: finding github.com/Azure/azure-sdk-for-go v27.0.0+incompatible
go: finding github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: finding github.com/hashicorp/go-uuid v1.0.1
go: finding github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: finding gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: finding github.com/modern-go/reflect2 v1.0.1
go: finding github.com/rjeczalik/notify v0.9.2
go: finding golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: finding github.com/googleapis/gax-go/v2 v2.0.4
go: finding google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: finding google.golang.org/grpc v1.20.1
go: finding go.opencensus.io v0.21.0
go: finding github.com/mitchellh/go-homedir v1.1.0
go: finding golang.org/x/crypto v0.0.0-20190923035154-9ee001bba392
go: finding golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: finding github.com/golang/protobuf v1.3.1
go: finding gopkg.in/ini.v1 v1.48.0
go: finding golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: finding github.com/hashicorp/golang-lru v0.5.1
go: finding gopkg.in/jcmturner/aescts.v1 v1.0.1
go: finding gopkg.in/jcmturner/rpc.v1 v1.1.0
go: finding golang.org/x/text v0.3.2
go: finding golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: finding gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: finding contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: finding github.com/gogo/protobuf v1.2.1
go: finding github.com/marstr/guid v1.1.0
go: finding github.com/satori/go.uuid v1.2.0
go: finding github.com/census-instrumentation/opencensus-proto v0.2.0
go: finding golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: finding github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: finding github.com/cheggaaa/pb v1.0.28
go: finding github.com/mattn/go-runewidth v0.0.4
go: finding github.com/coreos/etcd v3.3.12+incompatible
go: finding github.com/djherbis/atime v1.0.0
go: finding github.com/elazarl/go-bindata-assetfs v1.0.0
go: finding github.com/golang/snappy v0.0.1
go: finding github.com/gorilla/handlers v1.4.0
go: finding github.com/gorilla/rpc v1.2.0
go: finding github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: finding github.com/klauspost/readahead v1.3.0
go: finding github.com/klauspost/reedsolomon v1.9.1
go: finding github.com/klauspost/cpuid v1.2.1
go: finding github.com/miekg/dns v1.1.8
go: finding github.com/minio/dsync/v2 v2.0.0
go: finding github.com/minio/lsync v1.0.1
go: finding github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: finding github.com/hashicorp/vault v1.1.0
go: finding github.com/hashicorp/errwrap v1.0.0
go: finding github.com/hashicorp/go-cleanhttp v0.5.1
go: finding github.com/hashicorp/go-multierror v1.0.0
go: finding github.com/hashicorp/go-retryablehttp v0.5.3
go: finding github.com/hashicorp/go-rootcerts v1.0.1
go: finding github.com/hashicorp/hcl v1.0.0
go: finding github.com/pierrec/lz4 v2.0.5+incompatible
go: finding github.com/hashicorp/go-sockaddr v1.0.2
go: finding github.com/ryanuber/go-glob v1.0.0
go: finding github.com/mitchellh/mapstructure v1.1.2
go: finding github.com/ncw/directio v1.0.5
go: finding github.com/minio/sio v0.2.0
go: finding github.com/coredns/coredns v1.4.0
go: finding github.com/eclipse/paho.mqtt.golang v1.1.2-0.20190322152051-20337d8c3947
go: finding github.com/go-sql-driver/mysql v1.4.1
go: finding github.com/gomodule/redigo v2.0.0+incompatible
go: finding github.com/lib/pq v1.0.0
go: finding github.com/nats-io/nats.go v1.8.0
go: finding github.com/nats-io/nkeys v0.0.2
go: finding github.com/nats-io/nuid v1.0.1
go: finding github.com/nats-io/stan.go v0.4.5
go: finding github.com/nsqio/go-nsq v1.0.7
go: finding github.com/pkg/errors v0.8.1
go: finding github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: finding github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: finding gopkg.in/Shopify/sarama.v1 v1.20.0
go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/eapache/go-resiliency v1.2.0
go: finding github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: finding github.com/eapache/queue v1.1.0
go: finding github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: finding gopkg.in/olivere/elastic.v5 v5.0.80
go: finding github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: finding github.com/fatih/structs v1.1.0
go: finding github.com/tidwall/gjson v1.2.1
go: finding github.com/tidwall/match v1.0.1
go: finding github.com/tidwall/pretty v1.0.0
go: finding gopkg.in/yaml.v2 v2.2.2
go: finding github.com/klauspost/pgzip v1.2.1
go: finding github.com/klauspost/compress v1.5.0
go: finding github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: finding github.com/alecthomas/participle v0.2.1
go: finding github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: finding git.apache.org/thrift.git v0.12.0
go: finding github.com/pkg/profile v1.3.0
go: finding github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: finding github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: finding github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: finding github.com/prometheus/common v0.2.0
go: finding github.com/matttproud/golang_protobuf_extensions v1.0.1
go: finding github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: finding github.com/rs/cors v1.6.0
go: finding go.uber.org/atomic v1.3.2
go: finding gopkg.in/ldap.v3 v3.0.3
go: finding gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
# github.com/minio/minio/cmd
cmd/iam.go:472:24: cannot assign to struct field policyMap[k].Users in map
cmd/iam.go:477:25: cannot assign to struct field policyMap[k].Groups in map
make: *** [build] Error 2

```

--- a/cmd/admin-handlers.go
+++ b/cmd/admin-handlers.go
@@ -1329,6 +1329,25 @@ func (a adminAPIHandlers) AddUser(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// InfoCannedPolicy - GET /minio/admin/v1/info-canned-policy?name={policyName}
+func (a adminAPIHandlers) InfoCannedPolicy(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "InfoCannedPolicy")
+
+	objectAPI := validateAdminReq(ctx, w, r)
+	if objectAPI == nil {
+		return
+	}
+
+	data, err := globalIAMSys.InfoPolicy(mux.Vars(r)["name"])
+	if err != nil {
+		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+		return
+	}
+
+	w.Write(data)
+	w.(http.Flusher).Flush()
+}
+
 // ListCannedPolicies - GET /minio/admin/v1/list-canned-policies
 func (a adminAPIHandlers) ListCannedPolicies(w http.ResponseWriter, r *http.Request) {
 	ctx := newContext(r, w, "ListCannedPolicies")

--- a/cmd/admin-router.go
+++ b/cmd/admin-router.go
@@ -91,6 +91,9 @@ func registerAdminRouter(router *mux.Router, enableConfigOps, enableIAMOps bool)
 		adminV1Router.Methods(http.MethodPut).Path("/set-user-status").HandlerFunc(httpTraceHdrs(adminAPI.SetUserStatus)).
 			Queries("accessKey", "{accessKey:.*}").Queries("status", "{status:.*}")
 
+		// Info policy IAM
+		adminV1Router.Methods(http.MethodGet).Path("/info-canned-policy").HandlerFunc(httpTraceHdrs(adminAPI.InfoCannedPolicy)).Queries("name", "{name:.*}")
+
 		// Remove policy IAM
 		adminV1Router.Methods(http.MethodDelete).Path("/remove-canned-policy").HandlerFunc(httpTraceHdrs(adminAPI.RemoveCannedPolicy)).Queries("name", "{name:.*}")
 

--- a/cmd/iam.go
+++ b/cmd/iam.go
@@ -429,6 +429,23 @@ func (sys *IAMSys) DeletePolicy(policyName string) error {
 	return err
 }
 
+// InfoPolicy - expands the canned policy into its JSON structure.
+func (sys *IAMSys) InfoPolicy(policyName string) ([]byte, error) {
+	objectAPI := newObjectLayerFn()
+	if objectAPI == nil {
+		return nil, errServerNotInitialized
+	}
+
+	sys.RLock()
+	defer sys.RUnlock()
+
+	v, ok := sys.iamPolicyDocsMap[policyName]
+	if !ok {
+		return nil, errNoSuchPolicy
+	}
+	return json.Marshal(v)
+}
+
 // ListPolicies - lists all canned policies.
 func (sys *IAMSys) ListPolicies() (map[string][]byte, error) {
 	objectAPI := newObjectLayerFn()
@@ -581,6 +598,7 @@ func (sys *IAMSys) GetUserInfo(name string) (u madmin.UserInfo, err error) {
 	if sys.usersSysType != MinIOUsersSysType {
 		return madmin.UserInfo{
 			PolicyName: sys.iamUserPolicyMap[name].Policy,
+			MemberOf:   sys.iamUserGroupMemberships[name].ToSlice(),
 		}, nil
 	}
 
@@ -892,15 +910,15 @@ func (sys *IAMSys) GetGroupDescription(group string) (gd madmin.GroupDesc, err e
 		policy = ps[0]
 	}
 
-	sys.RLock()
-	defer sys.RUnlock()
-
 	if sys.usersSysType != MinIOUsersSysType {
 		return madmin.GroupDesc{
 			Name:   group,
 			Policy: policy,
 		}, nil
 	}
+
+	sys.RLock()
+	defer sys.RUnlock()
 
 	gi, ok := sys.iamGroupsMap[group]
 	if !ok {

--- a/pkg/madmin/policy-commands.go
+++ b/pkg/madmin/policy-commands.go
@@ -24,6 +24,31 @@ import (
 	"net/url"
 )
 
+// InfoCannedPolicy - expand canned policy into JSON structure.
+func (adm *AdminClient) InfoCannedPolicy(policyName string) ([]byte, error) {
+	queryValues := url.Values{}
+	queryValues.Set("name", policyName)
+
+	reqData := requestData{
+		relPath:     "/v1/info-canned-policy",
+		queryValues: queryValues,
+	}
+
+	// Execute GET on /minio/admin/v1/info-canned-policy
+	resp, err := adm.executeMethod("GET", reqData)
+
+	defer closeResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, httpRespToErrorResponse(resp)
+	}
+
+	return ioutil.ReadAll(resp.Body)
+}
+
 // ListCannedPolicies - list all configured canned policies.
 func (adm *AdminClient) ListCannedPolicies() (map[string][]byte, error) {
 	reqData := requestData{

